### PR TITLE
processName no longer caches lookups.

### PR DESCRIPTION
**Pull request by nkirsch**
I'm shocked we didn't notice this earlier; this essentially does a pid->comm lookup once,
and never again.

We are caching process information in `gather`, so it's OK to pay the lookup penalty here as needed.

### CODE REVIEW

File: ftrace/ftrace.go [hunk 1]
```diff
@@ -22,12 +22,11 @@ import (
...
-	fp                   FileProvider
-	eventTypes           map[int]*EventType
-	selectCases          []reflect.SelectCase
-	cachedProcessNames   map[int]string
-	isCachedProcessNames bool
-	cachedKallsyms       map[uint64]string
+	fp                 FileProvider
+	eventTypes         map[int]*EventType
+	selectCases        []reflect.SelectCase
+	cachedProcessNames map[int]string
```
Comment (tqorange) [inline ftrace/ftrace.go]: `cachedProcessNames` no longer used if you want to remove it.

File: ftrace/ftrace.go [hunk 2]
```diff
@@ -167,27 +166,29 @@ func (f *Ftrace) Capture(callback func(Events)) {
 }
 
 func (f *Ftrace) processName(pid int) string {
-	if !f.isCachedProcessNames {
-		f.isCachedProcessNames = true
-		processNameFile, err := f.fp.ReadFtraceFile("saved_cmdlines")
+	// Note that if this were set, we'd only do this once.
```
Comment (tqorange) [inline ftrace/ftrace.go]: I assume this comment was pertaining to `isCachedProcessNames`, if so, the comment can be removed.

File: ftrace/ftrace.go [hunk 3]
```diff
@@ -167,27 +166,29 @@ func (f *Ftrace) Capture(callback func(Events)) {
...
-			}
-			f.cachedProcessNames[p] = v[1]
+
+		if pid == p {
+			return v[1]
 		}
 	}
 
-	return f.cachedProcessNames[pid]
+	return ""
```
Comment (tqorange) [inline ftrace/ftrace.go:186]: The caller of this returns func will turn `""` into `<...>` which then Gather special handles. Could clean it up here if desired.

**Review by tqorange — APPROVED** [review]
FWIW, I did notice this during my Gather testing days. I probably just didn't have enough domain language acquired to describe it well-enough back then. 😕

File: ftrace/ftrace.go (hunk 3 above)
Comment (nkirsch) [reply ftrace/ftrace.go:186]: Thanks for pointing that out; I'll leave it for now.

File: ftrace/ftrace.go (hunk 2 above)
Comment (nkirsch) [reply ftrace/ftrace.go]: 👍

File: ftrace/ftrace.go (hunk 1 above)
Comment (nkirsch) [reply ftrace/ftrace.go]: Done, thanks.

**Pull request merged**